integration/ratelimit_test.go

```diff
@@ -0,0 +1,46 @@
+package integration
+
+import (
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/containous/traefik/integration/try"
+	"github.com/go-check/check"
+	checker "github.com/vdemeester/shakers"
+)
+
+type RateLimitSuite struct {
+	BaseSuite
+	ServerIP string
+}
+
+func (s *RateLimitSuite) SetUpSuite(c *check.C) {
+	s.createComposeProject(c, "ratelimit")
+	s.composeProject.Start(c)
+
+	s.ServerIP = s.composeProject.Container(c, "nginx1").NetworkSettings.IPAddress
+}
+
+func (s *RateLimitSuite) TestSimpleConfiguration(c *check.C) {
+	file := s.adaptFile(c, "fixtures/ratelimit/simple.toml", struct {
+		Server1 string
+	}{s.ServerIP})
+	defer os.Remove(file)
+
+	cmd, _ := s.cmdTraefik(withConfigFile(file))
+	err := cmd.Start()
+	c.Assert(err, checker.IsNil)
+	defer cmd.Process.Kill()
+
+	err = try.GetRequest("http://127.0.0.1:80/", 500*time.Millisecond, try.StatusCodeIs(http.StatusOK))
+	c.Assert(err, checker.IsNil)
+	err = try.GetRequest("http://127.0.0.1:80/", 500*time.Millisecond, try.StatusCodeIs(http.StatusOK))
+	c.Assert(err, checker.IsNil)
+	err = try.GetRequest("http://127.0.0.1:80/", 500*time.Millisecond, try.StatusCodeIs(http.StatusTooManyRequests))
+	c.Assert(err, checker.IsNil)
+
+	time.Sleep(3 * time.Second)
+	err = try.GetRequest("http://127.0.0.1:80/", 500*time.Millisecond, try.StatusCodeIs(http.StatusOK))
+	c.Assert(err, checker.IsNil)
+}
```

Comment:
Maybe can you set lesser values to `frontends.frontend1.ratelimit.rateset.rateset1` (average :4 and burst 5?) in the toml file and check if this rateset is respected too in the `TestSimpleConfiguration` test or in another one?

Reply:
👍 You're right.  Its a good idea to test that first time period too.